### DISCUSSION TITLE
Is it possible to monitor Floating Point Unit?

### DISCUSSION THREAD

**Discussion by abhinand5** [opcm/pcm#255] [Q&A] [answered]
I just wanted to ask if it is possible to count Floating Point Events using this tool as I couldn't find anything from the readme about this.

**Comment by opcm** [comment]
PCM supports custom event programming. You can check which floating point events are suported by your CPU using pmu-query.py and then program them using pcm-core utility.

**Comment by abhinand5** [comment]
Thanks for your quick response.

> I am new to this, so please bear with me. 

### I have 3 questions for you: 
1. I compiled the WinMSRDriver into msr.sys and put it inside the same directory as precompiled binaries. Getting the following error could you please help me? 
    ![image](https://user-images.githubusercontent.com/35622449/100847935-73fb5a00-34a6-11eb-9891-cb11e3b07a16.png)

2. Running the python script also gives an error? 
    ![image (1)](https://user-images.githubusercontent.com/35622449/100848466-29c6a880-34a7-11eb-9c85-bea7cf559217.png)

3. Should I customize the source code inside `PCM-Core_Win` folder or the `pcm-core.cpp` file in main directory? It would be great if you can let me know how to customize the core to include the counter I need.

**Comment by opcm** [comment, marked as answer]
Q1: did you sign the driver? This [link](https://docs.microsoft.com/en-us/windows-hardware/drivers/develop/signing-a-driver#:~:text=In%20order%20to%20sign%20a,by%20a%20trusted%20root%20authority.) can be helpful. Alternatively use the third-party winring driver as described in the windows how-to.

Q2: thanks for reporting this issue! For the Windows platform please replace ['pcm-core.exe -c'] with ['pcm-core.exe', '-c'] in pmu-query.py

Q3: no changes are required. With pmu-query.py you should be abe to search for floating point events (search for "FP") and then pass the event descriptions to pcm-core.exe

For example for my CPU:

`.\pcm-core.exe -e cpu/umask=0x02,event=0xC7,name=FP_ARITH_INST_RETIRED.SCALAR_SINGLE/`

**Comment by abhinand5** [comment]
Thanks a lot @opcm 

IT WORKED!!! 😃

**Comment by abhinand5** [comment]
@opcm 
Just a quick question. What do the letters in the output mean? `K, M, G, B` etc...

**Comment by opcm** [comment]
these are multpliers
K: *1024
M: *1024 *1024
etc
